### PR TITLE
Import newest versions of cryptoki and tss-esapi

### PR DESCRIPTION
**Pull request by hug-dev**
Imported as git dependencies to test cross-compilation of other targets.
The `Dockerfile` has been updated and changed to be built just to be tested on this CI. We can update the Docker image if that works.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,10 +26,10 @@ toml = "0.5.8"
 serde = { version = "1.0.123", features = ["derive"] }
 env_logger = "0.8.3"
 log = { version = "0.4.14", features = ["serde"] }
-cryptoki = { version = "0.1.0", optional = true, features = ["psa-crypto-conversions"] }
+cryptoki = { git = "https://github.com/parallaxsecond/rust-cryptoki", rev = "850b826b631df354553bf62757f35cd394b3dfff", optional = true, features = ["psa-crypto-conversions"] }
```
Comment (anta5010) [inline Cargo.toml:29]: Why not crate 0.1.1?

Comment (hug-dev) [reply Cargo.toml:29]: `0.1.1` does not contain https://github.com/parallaxsecond/rust-cryptoki/pull/18 which has some changes needed to fix #379

File: e2e_tests/docker_image/Dockerfile [hunk 2]
```diff
@@ -102,10 +99,6 @@ ENV PATH="/root/.cargo/bin:${PATH}"
 COPY _exec_wrapper /usr/local/bin/
 RUN ls /opt/rust/bin | xargs -n1 -I% ln -s /usr/local/bin/_exec_wrapper /usr/local/bin/$(basename %)
 
-# Adding the targets for the cross-compilation tests
-RUN rustup target add armv7-unknown-linux-gnueabihf
-RUN rustup target add aarch64-unknown-linux-gnu
-
 # Add users for multitenancy tests
```
Comment (hug-dev) [inline e2e_tests/docker_image/Dockerfile:102]: This `Dockerfile` can be uploaded so that it uses a bit less space 👌

Comment (ionut-arm) [reply e2e_tests/docker_image/Dockerfile:102]: You can remove the changes here, I've added them in my PR and the image has already been published! You just need to re-run the CI

**Review by ionut-arm — APPROVED** [review]

**Pull request merged**